src/middleware.ts

```diff
@@ -248,7 +281,11 @@ export const persist = <S extends State>(
       blacklist.forEach((key) => delete state[key])
     }
 
-    return storage?.setItem(name, await serialize({ state, version }))
+    if (storage) {
+      toThenable(serialize)({ state, version }).then((serializedValue) => {
```

Comment:
```suggestion
      return toThenable(serialize)({ state, version }).then((serializedValue) => {
```
Not sure if the `storage?.setItem` should also be returned since it's possibly async? 🤔

Reply:
In addition to it, my suggestion is to define `const thenabledSerialize = toThenable(serialize)` outside `setItem`, and

```js
if (storage) {
  return thenabledSerialize({ state, version })
    .then((serializedValue) => storage.setItem(name, serializedValue))
    .catch((e) => { throw e })
}
```

Reply:
As L259 checks `storage`, we don't need `if`.

Reply:
Ok, I did it. is there any reason to define it outside setItem?

Reply:
I think we still need `if (storage) ` because the function not immediately run after `L259`.
and there may be someone make it `undefined`. before calling setItem(). 
at least typescript will not complain about it.

Reply:
and yeah, we should move `if (storage)` inside `then` handler